### ISSUE TITLE
Symmetrical Simulation Freezes at Point of Reversal

### ISSUE THREAD

**Issue by pjbollinger** [DeviaVir/zenbot#752]
There is an issue with the `sim` command when using the `--symmetrical` option. It will simulate the history of trades like normal. However, once it reaches the point of reversing, it appears to get caught in a loop. I used mongoreplay to [capture this log (an excerpt of the full log)](https://gist.github.com/pjbollinger/7d8fcb4ef3c5aba2dc0a4ea4e9fd080d) and a [snippet at time of reversal](https://gist.github.com/pjbollinger/60c13387d75260e368394d3460eb18b6). There is a lot of data in there but you can see at the point of reversal there a couple of `find` orders in a row and then starts alternating `find` and `getMore` orders. After the reversal, it keeps repeating the same `find` and `getMore` orders though.

While debugging #750, I found that the program kept entering [this logic of the code](https://github.com/carlos8f/zenbot/blob/master/commands/sim.js#L187), which I think makes sense. However, the cursor does not appear to be moving correctly. I will be investigating this but wanted to share my findings so far.

**Comment by pjbollinger** [comment]
Here is some data of the queries around the code that identifies the original data is complete and it begins the reversal process. Notice how the `$lt` selector for the time freezes at the end.

Query (`opts`) at point of entering [`if (!trades.length) { if (so.symmetrical && !reversing) {...}}`](https://github.com/carlos8f/zenbot/blob/db9a5491f80be6598a578f035532ab5b66e6e99a/commands/sim.js#L206):
```
{ query:
  { selector: 'gdax.ETH-USD',
    time: { '$gt': 1511632608799 } },
  sort: { time: 1 },
  limit: 1000 }
```

Query (`opts`) at point of entering [`if (so.symmetrical && reversing) {...}`](https://github.com/carlos8f/zenbot/blob/db9a5491f80be6598a578f035532ab5b66e6e99a/commands/sim.js#L214) the first time:
```
{ query: 
   { selector: 'gdax.ETH-USD',
     time: { '$lt': 1511632608799, '$gte': 1511373600000 } },
  sort: { time: -1 },
  limit: 1000 }
```

Query (`opts`) at point of entering [`if (so.symmetrical && reversing) {...}`](https://github.com/carlos8f/zenbot/blob/db9a5491f80be6598a578f035532ab5b66e6e99a/commands/sim.js#L214) the second time:
```
{ query: 
   { selector: 'gdax.ETH-USD',
     time: { '$lt': 1511632981090, '$gte': 1511373600000 } },
  sort: { time: -1 },
  limit: 1000 }
```

Query (`opts`) at point of entering [`if (so.symmetrical && reversing) {...}`](https://github.com/carlos8f/zenbot/blob/db9a5491f80be6598a578f035532ab5b66e6e99a/commands/sim.js#L214) the third time and forever after:
```
{ query: 
   { selector: 'gdax.ETH-USD',
     time: { '$lt': 1511632980913, '$gte': 1511373600000 } },
  sort: { time: -1 },
  limit: 1000 }
```

**Comment by pjbollinger** [comment]
I found the issue is caused by the cursor not properly adjusting from the reversal of time.

Here is a screenshot of a simulation HTML report after the fix #753:
![screen shot 2017-11-26 at 3 57 23 pm](https://user-images.githubusercontent.com/5209474/33244892-76330d2e-d2cd-11e7-8846-b8d53027a4bf.png)

When running, I recommend using `--min_periods=0` to achieve near perfect symmetry.
